documentation/README.md

```diff
@@ -0,0 +1,32 @@
+# Documentation File
+
+The documentation for tablecloth is in file `documentation.md`. The goal is to have a single file that can be used to generate documentation in both OCaml and ReasonML format.
+
+Here is the general pattern for function documentation:
+
+<pre lang="no-highlight"><code>## square
+
+```ocaml
+ val square : int -> int 
+```
+
+```reason
+let square: int => int;
+```
+
+Description of function goes here. To mark inline code as OCaml,
+follow it with {:.ocaml}; to mark it as ReasonML, follow it
+with {:.reason}. (This is a convention from kramdown
+https://kramdown.gettalong.org/) Example:
+
+`square n`{:.ocaml} `square(n)`{:.reason} returns the square of `n`.
```

Comment:
What's the goal of this?

It doesn't seem to work in the github preview, fyi.

Reply:
The goal is to be able to generate OCaml and Reason documentation by having a program that extracts the relevant information, or to generate an HTML page with some JavaScript magic that lets you display the code in your desired language. (Github pages does recognize the attributes, according to https://stackoverflow.com/questions/23226224/inline-code-syntax-highlighting-in-github-markdown)

Reply:
Gotcha. Now that I see how this is going, I think we should change strategies. Let's just get the mlis documented, and worry about html later.

Reply:
OK; then does that mean the mli comments should be in ocamldoc format? If so, that leaves ReasonML users out in the cold.

Reply:
They don't need to be in ocamldoc format either. The format doesn't matter. The important thing is getting content into the mli files that explain what the functions do. (And that show up in editors via merlin, vscode lsp, etc - but this can be a separate step).

Once we're in a good shape there, we can start thinking about how to reformat for best reuse, etc.

Reply:
Oh, OK. Then if you have no objection , I will keep this markdown format in the `.mli` file, unless you have a better suggestion for handling both ReasonML and OCaml.

Reply:
Sure, sounds good. Thanks!